### PR TITLE
Added integration test for GetExchangeRate operation

### PR DESCRIPTION
**Pull request by kkovarik**
Added simple end-to-end test. Test does launch entire openhub-ri application, with http client request is sent, and live target REST service (fixer.io) is replaced with mocked http server.

### CHANGES
* added dependency to openhub-web to start tests with web context & embedded servlet container (see related PR to openhub-framework which enables that)
* added simple parent AbstractIntegrationTest. does contain prepared
* added "happy scenario" test implementation for GetExchangeRate operation, see GetExchangeRateIntegrationTest

### FUTHER STEPS
* AbstractIntegrationTest could be transferred to openhub-framework with improved mock server (for example with not static port)

### SEE ALSO
* related openhub-framework PR: https://github.com/OpenWiseSolutions/openhub-framework/pull/70 , this changes depend on it
* used mock server: https://github.com/mkotsur/restito

### CODE REVIEW

File: openhub-ext/pom.xml [hunk 1]
```diff
@@ -48,6 +47,10 @@
             <groupId>org.openhubframework</groupId>
             <artifactId>openhub-components</artifactId>
         </dependency>
+        <dependency>
+            <groupId>org.openhubframework</groupId>
```
Comment (pjuza) [inline openhub-ext/pom.xml]: Why not for test scope only?

File: openhub-ext/src/test/java/org/openhubframework/openhub/ri/AbstractIntegrationTest.java [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+        "test.camel.initAllRoutes = true"
+})
+@SpringBootTest(
+        classes = {
+                TestConfig.class,
+                TestWebConfig.class
+        },
+        webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT
+)
+public abstract class AbstractIntegrationTest extends AbstractRITest {
```
Comment (pjuza) [inline openhub-ext/src/test/java/org/openhubframework/openhub/ri/AbstractIntegrationTest.java]: I would expect that parent class for web integration tests will be in OHF in test module, like AbstractTest or AbstractDbTest. It meens also add dependency to restito ...
Another project will have to creates identical/very similar classes, why don't add it to OHF?

**Review by pjuza — CHANGES_REQUESTED** [review]
Please try to think about if parent class for integration tests don't move to OHF?

File: openhub-ext/pom.xml (hunk 1 above)
Comment (kkovarik) [reply openhub-ext/pom.xml]: Right, will change, thanks.

**Comment by kkovarik** [comment]
@pjuza: Updated to correspond with new changes in openhub-framework - added tomcat dependency & custom component scan config.

File: openhub-ext/pom.xml (hunk 1 above)
Comment (kkovarik) [reply openhub-ext/pom.xml]: Done.

**Comment by kkovarik** [comment]
Closed, served its purpose, will be added directly to the target branch,

**Pull request closed without merging**